src/libraries/System.Runtime.Loader/tests/AssemblyExtensionsTest.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Reflection.Metadata
+{
+    public class AssemblyExtensionsTest
+    {
+        class NonRuntimeAssembly : Assembly
+        {
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/45689", platforms: TestPlatforms.AnyUnix | TestPlatforms.OSX, runtimes: TestRuntimes.Mono)]
+        public static void ApplyUpdateInvalidParameters()
+        {
+            // Dummy delta arrays
+            var metadataDelta = new byte[20];
+            var ilDelta = new byte[20];
+
+            // Assembly can't be null
+            Assert.Throws<ArgumentNullException>("assembly", () =>
+                AssemblyExtensions.ApplyUpdate(null, new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
+
+            // Tests fail on non-runtime assemblies
+            Assert.Throws<ArgumentException>(() =>
+                AssemblyExtensions.ApplyUpdate(new NonRuntimeAssembly(), new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
+
+            // Tests that this assembly isn't not editable
+            Assert.Throws<InvalidOperationException>(() =>
+                AssemblyExtensions.ApplyUpdate(typeof(AssemblyExtensions).Assembly, new ReadOnlySpan<byte>(metadataDelta), new ReadOnlySpan<byte>(ilDelta), ReadOnlySpan<byte>.Empty));
```

Comment:
I'd weakly prefer if this was in a separate test - for now all assemblies are potentially editable in Mono (I'll revisit once we have a plan for https://github.com/dotnet/runtime/issues/47274).

It's ok if this is merged as is - I'll move it out when I rebase https://github.com/dotnet/runtime/pull/48380.

Reply:
You may want to consider not allowing System.Private.CoreLib.dll (which the AssemblyExtensions class is in) to be edited like coreclr does or what it considers "system" assemblies.

Reply:
>  You may want to consider not allowing System.Private.CoreLib.dll (which the AssemblyExtensions class is in) to be edited like coreclr does or what it considers "system" assemblies.

oh I went chasing down the wrong branch in `Module::IsEditAndContinueCapable` - thought this was the debug bits, not `PEFile::IsSystem`.  Yea it makes sense to not allow edits to CoreLib.